targets/TARGET_NXP/TARGET_LPC15XX/can_api.c

```diff
@@ -187,13 +238,53 @@ void can_irq_free(can_t *obj) {
 
 // Clear or set a irq
 void can_irq_set(can_t *obj, CanIrqType type, uint32_t enable) {
+    uint32_t mask_enable;
+    switch (type) {
+       case IRQ_RX:
+           mask_enable = IRQ_ENABLE_RX;
+           break;
+       case IRQ_TX:
+           mask_enable = IRQ_ENABLE_TX;
+           break;
+       case IRQ_BUS:
+           mask_enable = IRQ_ENABLE_BE;
+           break;
+       case IRQ_PASSIVE:
+           mask_enable = IRQ_ENABLE_EP;
+           break;
+       case IRQ_ERROR:
+           mask_enable = IRQ_ENABLE_EW;
+           break;
+       default:
+           return;
+    }
+    
+    if (enable) {
+        enabled_irqs = enabled_irqs | mask_enable;
+    } else {
+        enabled_irqs = enabled_irqs & ~mask_enable;
+    }
+
     // Put CAN in Reset Mode and enable interrupt
     can_disable(obj);
-    if (enable == 0) {
-        LPC_C_CAN0->CANCNTL &= ~(1UL << 1 | 1UL << 2);
+    if (!(enabled_irqs && IRQ_ENABLE_ANY)) {
```

Comment:
Is this ```operator &&``` (logical AND) intentional here?  I think this should be ```operator &``` (Bitwise AND).